main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -702,6 +705,19 @@ object Safety {
     terms.flatMap(visitPat(_, loc))
   }
 
+  /**
+   *  Checks whether an unchecked cast is allowed to use or not
+   *  If not it will return an 'IncorrectSafetySignature'
+   */
+  private def isUncheckedCastAllowed(loc: SourceLocation)(implicit flix: Flix): List[SafetyError] = loc.source.input match {
+    case _: Input.Text => Nil
+    case _: Input.PkgFile => Nil
+    case _: Input.TxtFile => if (flix.safe) IncorrectSafetySignature(loc) :: Nil else Nil
```

Comment:
I suppose we need another variant for the standard library.

Reply:
I will handle this in the next PR where the Input types will be extended.